safekeeper/src/timeline.rs

```diff
@@ -116,6 +116,9 @@ pub struct SharedState {
     /// when tli is inactive instead of having this flag.
     active: bool,
     last_removed_segno: XLogSegNo,
+    /// True if local and remote deletion has been done, allows to skip visiting
+    /// s3 on retries.
+    deleted: bool,
```

Comment:
Let's rename this to `fully_deleted` maybe, to hint that deletion is not instant and can take some time.

Also, I think it will be better to move this flag to `timelines_global_map.rs` eventually, so maybe let's add a todo for that here. I'll create an issue for better deletion handling meanwhile.

Reply:
Renamed. WDYT about renaming 'cancelled' to 'deleted' then?

Reply:
Not now, maybe later. Currently `cancelled` is another flag which is set as early as possible after receiving deletion request.